app/managers/session_manager.test.ts

```diff
@@ -0,0 +1,204 @@
+import NetworkManager from '@managers/network_manager';
+import WebsocketManager from '@managers/websocket_manager';
+import {deleteFileCache, deleteFileCacheByDir} from '@utils/file';
+import {isMainActivity} from '@utils/helpers';
+
+import {SessionManager as SessionManagerClass} from './session_manager';
+
+jest.mock('@react-native-cookies/cookies', () => ({
+    get: jest.fn(),
+    clearByName: jest.fn(),
+    flush: jest.fn(),
+}));
+jest.mock('expo-image');
+jest.mock('@actions/app/global');
+jest.mock('@actions/remote/session');
+jest.mock('@database/manager', () => ({
+    getActiveServerUrl: jest.fn(),
+    getActiveServerDisplayName: jest.fn(),
+    destroyServerDatabase: jest.fn(),
+    deleteServerDatabase: jest.fn(),
+    serverDatabases: {},
+}));
+jest.mock('@i18n');
+jest.mock('@init/credentials');
+jest.mock('@init/launch');
+jest.mock('@init/push_notifications');
+jest.mock('@managers/network_manager');
+jest.mock('@managers/websocket_manager');
+jest.mock('@queries/app/servers');
+jest.mock('@queries/servers/user');
+jest.mock('@screens/navigation');
+jest.mock('@store/ephemeral_store');
+jest.mock('@utils/file');
+jest.mock('@utils/helpers');
+
+describe('SessionManager', () => {
+    const mockServerUrl = 'https://example.com';
+    const mockServerDisplayName = 'Example Server';
+    let appStateCallback: ((state: string) => void) | undefined;
+    let SessionManager: SessionManagerClass;
+
+    jest.mocked(isMainActivity).mockReturnValue(true);
+
+    jest.mocked(DatabaseManager.getActiveServerUrl).mockResolvedValue(mockServerUrl);
+    jest.mocked(DatabaseManager.getActiveServerDisplayName).mockResolvedValue(mockServerDisplayName);
+
+    (CookieManager.get as jest.Mock).mockResolvedValue({
+        cookie1: {name: 'cookie1'},
+        cookie2: {name: 'cookie2'},
+    });
+
+    jest.mocked(getAllServerCredentials).mockResolvedValue([{serverUrl: mockServerUrl, userId: 'user_id', token: 'token'}]);
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        AppState.currentState = 'active';
+        Platform.OS = 'ios';
+
+        (AppState.addEventListener as jest.Mock).mockImplementation((event, callback) => {
+            if (event === 'change' || event === 'focus' || event === 'blur') {
+                appStateCallback = callback;
+            }
+            return {remove: jest.fn()};
+        });
+
+        SessionManager = new SessionManagerClass();
+    });
+
+    describe('initialization', () => {
+        it('should construct with Android correctly', async () => {
+            Platform.OS = 'android';
+            const manager = new SessionManagerClass();
+            expect(manager).toBeDefined();
+            expect(AppState.addEventListener).toHaveBeenCalledWith('blur', expect.any(Function));
+            expect(AppState.addEventListener).toHaveBeenCalledWith('focus', expect.any(Function));
+        });
+    });
+
+    describe('initialization', () => {
```

Comment:
I've noticed there are two `describe` blocks with the title 'initialization'. Should we consolidate them into a single `describe` block?